gtsam/geometry/tests/testPose3.cpp

```diff
@@ -1016,6 +1016,33 @@ TEST(Pose3, TransformCovariance6) {
 TEST(Pose3, interpolate) {
   EXPECT(assert_equal(T2, interpolate(T2,T3, 0.0)));
   EXPECT(assert_equal(T3, interpolate(T2,T3, 1.0)));
+
+  // Trivial example: start at origin and move to (1, 0, 0) while rotating pi/2
+  // about z-axis.
+  Pose3 start;
+  Pose3 end(Rot3::Rz(M_PI_2), Point3(1, 0, 0));
+  // This interpolation is easy to calculate by hand.
+  double t = 0.5;
+  Pose3 expected0(Rot3::Rz(M_PI_4), Point3(0.5, 0, 0));
+  EXPECT(assert_equal(expected0, interpolate(start, end, t)));
```

Comment:
@dellaert I added this simple test to verify since it is easy to visualize and imaging: a body turning about Z as it moves from the origin to `(1, 0, 0)`.

The previous code fails with this result:
```
Not equal:
expected:
 R: [
	0.707107, -0.707107, 0;
	0.707107, 0.707107, 0;
	0, 0, 1
]
t: 0.5   0   0
actual:
 R: [
	0.707107, -0.707107, 0;
	0.707107, 0.707107, 0;
	0, 0, 1
]
t:       0.5 -0.207107         0
```

It doesn't make any sense why the body would off the X axis.

Reply:
This is because the interpolation is in *tangent* space. Straight lines in tangent space correspond to screw motions.

Reply:
Can you please provide a reference I can read up on?

Reply:
https://en.wikipedia.org/wiki/Slerp

The reference is the documentation of "interpolate". I think you expected something else, which is fine. I emailed with a different approach (interpolateRt). You can document the difference with interpolate<Pose3> there.